### PR TITLE
added tls-version-min

### PR DESCRIPTION
**Pull request by peterbeck**
added tls-version-min as discussed in #264. this is my first PR - I hope I've done it correct...

### CODE REVIEW

File: manifests/server.pp [hunk 1]
```diff
@@ -500,6 +507,7 @@
   String $verb                              = '',
   String $cipher                            = 'AES-256-CBC',
   String $tls_cipher                        = 'TLS-DHE-RSA-WITH-AES-256-GCM-SHA384:TLS-DHE-RSA-WITH-AES-256-CBC-SHA256:TLS-DHE-RSA-WITH-AES-128-GCM-SHA256:TLS-DHE-RSA-WITH-AES-128-CBC-SHA256',
+  String $tls_version_min                   = '',
```
Comment (bastelfreak) [inline manifests/server.pp]: Datatypes \o/

Comment (bastelfreak) [reply manifests/server.pp]: Can you change the logic so it defaults to undef instead of empty string? That would be the current best practice.

**Comment by bastelfreak** [comment]
Hi @peterbeck, thanks for the PR. I'm wondering if we should provide a secure default setting for this option, or if it should default to nothing. I'm not very familiar with openvpn so I don't know which openvpn versions support which TLS versions and how easy it is to implement this. Do you have any ideas?

----

Can you please add a unit test that verifies that the config file contains the correct line?

**Comment by peterbeck** [comment]
Hi @bastelfreak, thanks for your feedback. I think we should leave it to 'undef' by default not to break existing deployments, otherwise older (<2.3.4) clients [won't be able to connect ](https://community.openvpn.net/openvpn/ticket/401). But I'm also not that sure in this case.. I'll change the empty string and try to add a unit test (have to read first...)

**Comment by bastelfreak** [comment]
Alright, lets stick with undef.

**Comment by alexjfisher** [comment]
@peterbeck Have you had chance to look at making the change to use 'undef' and adding a test?

**Comment by peterbeck** [comment]
I'm just asking myself if this pull request is necessary, as this parameter could also be set with custom_options.... 
```
custom_options => {
  "tls-version-min" => "1.2"
}
```
so far it seems I was the only one confronted with issues while not setting tls-min-version. 
what do you think ?

**Comment by simondeziel** [comment]
@peterbeck, I see value in your PR and Optional[String] seems the way to go IMHO.

**Comment by Dan33l** [comment]
Any news in the PR ?

Now a rebase is needed.

**Comment by bastelfreak** [comment]
Hi people. This PR looks stalled, I'm going to close it. Please reopen and rebase it if you're still interested in it.

**Pull request closed without merging**